libres/tests/analysis/test_save_parameters.cpp

```diff
@@ -102,3 +116,88 @@ TEST_CASE("Write and read a matrix to enkf_fs instance",
+        int ensemble_size = 10;
+        // setting up a config node for a single parameter
+        auto config_node =
+            ensemble_config_add_gen_kw(ensemble_config, "TEST", false);
+        // create template file
+        std::ofstream templatefile("template");
+        templatefile << "{\n\"a\": <COEFF>\n}" << std::endl;
+        templatefile.close();
+
+        // create parameter_file
+        std::ofstream paramfile("param");
+        paramfile << "COEFF_A UNIFORM 0 1" << std::endl;
+        paramfile << "COEFF_B UNIFORM 0 1" << std::endl;
+        paramfile.close();
+
+        enkf_config_node_update_gen_kw(config_node, "not_important.txt",
+                                       "template", "param", nullptr, nullptr);
+
+        // Creates files on fs where nodes are stored.
+        // This is needed for the deserialization of the matrix, as the
+        // enkf_fs instance has to instantiate the files were things are
+        // stored.
+        enkf_node_type *node = enkf_node_alloc(config_node);
+        for (int i = 0; i < ensemble_size; i++) {
+            enkf_node_store(node, fs, {.report_step = 0, .iens = i});
+        }
+        enkf_node_free(node);
+
+        // set up ministep with one parmater and N realizations
+        local_ministep_type *ministep =
+            new local_ministep_type("not-important", nullptr);
+        ministep->add_active_data("TEST");
+        row_scaling_type *scaling =
+            local_ministep_get_or_create_row_scaling(ministep, "TEST");
+        int_vector_type *active_index = int_vector_alloc(ensemble_size, -1);
+        for (int i = 0; i < ensemble_size; i++) {
+            int_vector_iset(active_index, i, i);
+        }
+
+        // Create matrix and save as as the parameter defined in the ministep
+        matrix_type *A = matrix_alloc(2, ensemble_size);
+        for (int i = 0; i < ensemble_size; i++) {
+            matrix_iset(A, 0, i, double(i) / 10.0);
+            matrix_iset(A, 1, i, double(i) / 20.0);
+        }
+
+        std::vector<std::pair<matrix_type *, const row_scaling_type *>>
+            row_scaling_list;
+
+        row_scaling_list.push_back(
+            std::pair<matrix_type *, const row_scaling_type *>(A, scaling));
+        analysis::save_row_scaling_parameters(fs, ensemble_config, active_index,
+                                              ministep, row_scaling_list);
+
+        WHEN("loading parameters from enkf_fs") {
+            auto parmeter_matrices = analysis::load_row_scaling_parameters(
+                fs, ensemble_config, active_index, 0, ensemble_size, ministep);
+            THEN("Loading parameters yield the same matrix") {
+                for (int i = 0; i < row_scaling_list.size(); i++) {
+                    auto A = row_scaling_list[i].first;
+                    auto B = row_scaling_list[i].first;
+                    REQUIRE(matrix_equal(A, B));
+                }
+            }
+        }
+
+        //cleanup
```

Comment:
I think we should cleanup `scaling` as well with a call to `row_scaling_free`.

Reply:
The rowscaling object is handled by the ministep object, and destroyed when the ministep is deleted.